diff --git a/krb5-ticket-watcher/krb5-ticket-watcher.desktop b/krb5-ticket-watcher/krb5-ticket-watcher.desktop
index 2ba6c0c..6127c66 100644
--- a/krb5-ticket-watcher/krb5-ticket-watcher.desktop
+++ b/krb5-ticket-watcher/krb5-ticket-watcher.desktop
@@ -1,5 +1,4 @@
 [Desktop Entry]
-Encoding=UTF-8
 Exec=krb5-ticket-watcher -i 31
 Path=
 Type=Application
@@ -7,6 +6,7 @@ MimeType=
 Terminal=false
 Icon=krb5-ticket-watcher
 GenericName=Kerberos Ticket Watcher
+GenericName[ru]=Отслеживание билетов Kerberos
 Name=Kerberos Ticket Watcher
 X-KDE-StartupNotify=false
-Categories=System;Network
+Categories=System;Security;
