--- a/src/pwdialog.h
+++ b/src/pwdialog.h
@@ -21,6 +21,13 @@ public:
 		setModal(modal);
 		setAccessibleName(name);
 		setupUi(this);
+		QWidget *wnd = window();
+		if( wnd ) {
+		    wnd->setWindowFlags(wnd->windowFlags() | Qt::WindowStaysOnTopHint);
+		    wnd->show();
+		    wnd->setWindowState((wnd->windowState() & ~Qt::WindowMinimized) | Qt::WindowActive);
+		    wnd->raise();
+		}
 	}
 	
 	~PWDialog(){}
